### ISSUE TITLE
Support Non Interactive Clients just like Users

### ISSUE THREAD

**Issue by axos88** [auth0/auth0-authorization-extension#201]
One should be able to assign roles and permissions to a non-interactive client just like to a user. A rule could then be created to enrich the accessToken when requesting a client credential grant.

This make sense only after APIs are supported as targets for permissions and groups (although that limitation can be worked around)

**Comment by axos88** [comment]
#157

**Comment by valeriomazzeo** [comment]
@axos88 is there any workaround at the moment?

**Comment by axos88** [comment]
No, we wound up just giving all permissions to our non interactive clients, since all of them are in house, but we will probably implement our own PDP soon, in Q1 auth0 unofficially promised to have it ready Q2-Q3, but right now they still said that they don't have any official date set, and unofficially said it won't be anytime soon.
